diesel/src/pg/types/integers.rs

```diff
@@ -1,14 +1,77 @@
 use byteorder::{NetworkEndian, ReadBytesExt, WriteBytesExt};
 use std::io::prelude::*;
+use std::error::Error;
 
 use deserialize::{self, FromSql};
-use pg::Pg;
+use pg::{Pg, PgValue};
 use serialize::{self, IsNull, Output, ToSql};
 use sql_types;
 
+impl FromSql<sql_types::SmallInt, Pg> for i16 {
+    fn from_sql(value: Option<PgValue>) -> deserialize::Result<Self> {
+        let value = not_none!(value);
+        let mut bytes = value.bytes();        
+        debug_assert!(
+            bytes.len() <= 2,
+            "Received more than 2 bytes decoding i16. \
+             Was an Integer expression accidentally identified as SmallInt?"
+        );
+        debug_assert!(
+            bytes.len() >= 2,
+            "Received fewer than 2 bytes decoding i16. \
+             Was an expression of a different type accidentally identified \
+             as SmallInt?"
+        );
+        bytes
+            .read_i16::<NetworkEndian>()
+            .map_err(|e| Box::new(e) as Box<Error + Send + Sync>)
+    }
+}
+
+impl FromSql<sql_types::Integer, Pg> for i32 {
+    fn from_sql(value: Option<PgValue>) -> deserialize::Result<Self> {
+        let value = not_none!(value);
+        let mut bytes = value.bytes();
+        debug_assert!(
+            bytes.len() <= 4,
+            "Received more than 4 bytes decoding i32. \
+             Was a BigInteger expression accidentally identified as Integer?"
```

Comment:
Time to fix those ancient error message
```suggestion
             Was a BigInt expression accidentally identified as Integer?"
```